src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -393,17 +393,24 @@ def _resolve_policy_id(cmd, policy, policy_set_definition, client):
             policy_def = _get_custom_or_builtin_policy(cmd, client, policy)
             policy_id = policy_def.id
         else:
-            policy_set_def = _get_custom_or_builtin_policy(cmd, client, policy_set_definition, True)
+            policy_set_def = _get_custom_or_builtin_policy(cmd, client, policy_set_definition, None, None, True)
             policy_id = policy_set_def.id
     return policy_id
 
 
-def _get_custom_or_builtin_policy(cmd, client, name, for_policy_set=False):
+def _get_custom_or_builtin_policy(cmd, client, name, subscription_id=None, management_group=None, for_policy_set=False):
     from msrest.exceptions import HttpOperationError
     from msrestazure.azure_exceptions import CloudError
     ErrorResponseException = cmd.get_models('ErrorResponseException')
     policy_operations = client.policy_set_definitions if for_policy_set else client.policy_definitions
+
+    if cmd.supported_api_version(min_api='2018-03-01'):
+        enforce_mutually_exclusive(subscription_id, management_group)
+        if subscription_id:
+            client.config.subscription_id = subscription_id
```

Comment:
This is not supported in CLI. We do not currently allow the user to make API calls from another subscription without running `az account set --subscription`.